src/main/java/dev/waterdog/waterdogpe/network/rewrite/EntityMap.java

```diff
@@ -242,6 +242,16 @@ public boolean handle(EmoteListPacket packet) {
         return true;
     }
 
+    public boolean handle(NpcDialoguePacket packet) {
+        packet.setRuntimeEntityId(PlayerRewriteUtils.rewriteId(packet.getRuntimeEntityId(), this.rewrite.getEntityId(), this.rewrite.getOriginalEntityId()));
+        return true;
+    }
+
+    public boolean handle(NpcRequestPacket packet) {
```

Comment:
Since we started the rewrite of this packet, it should be also added to 313 fast-codec

Reply:
Fixed in 1c7e588